test/framework/deployment_helpers.go

```diff
@@ -179,51 +183,111 @@ type watchPodLogsInput struct {
 // retrieving the deployment, its pods, or setting up a log file. If there is an error with the log streaming itself,
 // that does not cause the test to fail.
 func watchPodLogs(ctx context.Context, input watchPodLogsInput) {
-	pods := &corev1.PodList{}
-	Expect(input.GetLister.List(ctx, pods, client.InNamespace(input.Namespace), client.MatchingLabels(input.Labels))).To(Succeed(), "Failed to list Pods for deployment %s", klog.KRef(input.Namespace, input.DeploymentName))
+	// Create informer to watch for pods matching input.
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(
```

Comment:
Solution looks good from my perspective.

@vincepri Are we missing a more elegant one?

Reply:
@chrischdi Talked a bit with Vince.

The current implementation creates an informerFactory for every Deployment that we watch.
As far as I know that also means that we have separate informers for every one of them.

One alternative that we see would be to create a CR cache with cache.New() (maybe similar to the GetClient func on ClusterProxy). Then we can pass down the cache and get/create informers via `GetInformer()`.

Alternatively we could als create single informerFactory in ClusterProxy, but I think it could be interesting to explore how we can use the CR cache for that.

Reply:
Thank you!

Reply:
Q. should the informer be something at the ClusterProxy level? (thinking about the self-hosted tests or other tests where we have many clusters to stream logs from)

Reply:
Using `cache.New` works. I tried to adjust to that. However, please see the question I added as comment.